misk-jobqueue-testing/src/main/kotlin/misk/jobqueue/FakeJobQueue.kt

```diff
@@ -0,0 +1,58 @@
+package misk.jobqueue
+
+import misk.tokens.TokenGenerator
+import java.time.Duration
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentLinkedDeque
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class FakeJobQueue @Inject constructor(
```

Comment:
I think it would be nice if library fakes had tests. Even better if we had a test that verifies `FakeJobQueue` behaves just like the real job queue using in the in memory SQS

Reply:
👍 I had 'em in the jobqueue exemplar, but I'll add here.

Reply:
@ryanhall07 added some tests